jsonschema/validators.py

```diff
@@ -79,7 +79,10 @@ def iter_errors(self, instance, _schema=None):
             if _schema is None:
                 _schema = self.schema
 
-            with self.resolver.in_scope(_schema.get(u"id", u"")):
+            scope = _schema.get(u"id")
+            if scope:
```

Comment:
Probably want `is not None`? Which might mean a missing test case? (an `id` of "" is probably meaningless, but I guess it means something reasonable relative to some other base URI.)